index.js

```diff
@@ -4,39 +4,22 @@ module.exports = {
   name: 'live-reload-middleware',
 
   contentFor: function(type) {
-    var liveReloadPort = process.env.EMBER_CLI_INJECT_LIVE_RELOAD_PORT;
-    var baseURL = process.env.EMBER_CLI_INJECT_LIVE_RELOAD_BASEURL;
+    var liveReloadScriptSrc = process.env.EMBER_CLI_INJECT_LIVE_RELOAD_SCRIPT_SRC;
 
-    if (liveReloadPort && type === 'head') {
-      return '<script src="' + baseURL + 'ember-cli-live-reload.js" type="text/javascript"></script>';
+    if (liveReloadScriptSrc && type === 'head') {
+      return '<script src="' + liveReloadScriptSrc + '" type="text/javascript"></script>';
     }
   },
-
-  dynamicScript: function(request) {
-    var liveReloadPort = process.env.EMBER_CLI_INJECT_LIVE_RELOAD_PORT;
-
-    return "(function() {\n " +
-           "var src = (location.protocol || 'http:') + '//' + (location.hostname || 'localhost') + ':" + liveReloadPort + "/livereload.js?snipver=1';\n " +
-           "var script    = document.createElement('script');\n " +
-           "script.type   = 'text/javascript';\n " +
-           "script.src    = src;\n " +
-           "document.getElementsByTagName('head')[0].appendChild(script);\n" +
-           "}());";
-  },
-
   serverMiddleware: function(config) {
-    var self = this;
-    var app = config.app;
     var options = config.options;
 
     if (options.liveReload !== true) { return; }
 
-    process.env.EMBER_CLI_INJECT_LIVE_RELOAD_PORT = options.liveReloadPort;
-    process.env.EMBER_CLI_INJECT_LIVE_RELOAD_BASEURL = options.baseURL; // default is '/'
+    var liveReloadHost = options.liveReloadHost || options.host;
```

Comment:
options.liveReloadHost has been implemented in https://github.com/ember-cli/ember-cli/commit/7a6a47e12df5c6fda7fc8c3cdfb8e3237bfbe848